### PR TITLE
add checking if modules is an empty collection in lambdify

### PR DESCRIPTION
**Pull request by NChechulin**
#### References to other Issues or PRs
Fixes #17737 

#### Brief description of what is fixed or changed
Added checking is `modules` is a collection with length 0.
Now `lambdify([x, y], expr, modules=[])` behaves same way as `lambdify([x, y], expr)`

#### Release Notes

* utilities
  * fix issue with incorrect behaviour of `Mod` when `modules = []`

### CODE REVIEW

File: sympy/utilities/lambdify.py [hunk 1]
```diff
@@ -737,7 +737,7 @@ def _lambdifygenerated(x):
     from sympy.core.symbol import Symbol
```
Comment (czgdp1807) [inline sympy/utilities/lambdify.py:737]:
```suggestion
    from sympy.core.symbol import Symbol
    from collections.abc import Iterable
```

File: sympy/utilities/lambdify.py [hunk 2]
```diff
@@ -737,7 +737,7 @@ def _lambdifygenerated(x):
...
 
     # If the user hasn't specified any modules, use what is available.
-    if modules is None:
+    if modules is None or hasattr(modules, '__len__') and len(modules) == 0:
```
Comment (czgdp1807) [inline sympy/utilities/lambdify.py:740]:
```suggestion
    if modules:
```
The above will be false if the `modules` is an empty container or `None`.

Comment (czgdp1807) [reply sympy/utilities/lambdify.py:740]: Or where possible use anything in the check except `hasattr` and stuff.

Comment (NChechulin) [reply sympy/utilities/lambdify.py:740]: Then it should be `if not modules`, right?

Comment (czgdp1807) [reply sympy/utilities/lambdify.py:740]: Oh yeah, right. I mixed it up. Though is there anything in the documentation which specifies what all types `modules` can take.

**Comment by oscarbenjamin** [comment]
Can you add a test?

**Comment by NChechulin** [comment]
@oscarbenjamin, I'm not familiar with the project structure yet, but I will definitely try. I will write about the results a few hours later.

**Comment by oscarbenjamin** [comment]
The test could go in `sympy/utilities/tests/test_lambdify.py`

File: sympy/utilities/tests/test_lambdify.py [hunk 3]
```diff
@@ -1335,3 +1335,11 @@ def test_scipy_special_math():
 
     cm1 = lambdify((x,), cosm1(x), modules='scipy')
     assert abs(cm1(1e-20) + 5e-41) < 1e-200
+
+def test_empty_modules():
```
Comment (oscarbenjamin) [inline sympy/utilities/tests/test_lambdify.py:1339]: There should be two blank lines between top level functions

**Comment by oscarbenjamin** [comment]
This PR looks good to me except that it has merge conflicts now.

Note that adding tests at the end of the `test_` file will often lead to merge conflicts. (If everyone adds tests at the end then any two PRs adding tests to the same file are guaranteed to conflict.)

**Comment by oscargus** [comment]
Closing this as it has been solved elsewhere.

**Pull request closed without merging**